SIGHTMACHINE-FORK.md

```diff
@@ -0,0 +1,105 @@
+# Sightmachine fork
+
+## Git Setup
+
+Keep the remote names as `sightmachine` and `facebook`
+
+```
+> git remote -v
+facebook        git@github.com:facebook/rocksdb (fetch)
+facebook        git@github.com:facebook/rocksdb (push)
+sightmachine    git@github.com:sightmachine/rocksdb (fetch)
+sightmachine    git@github.com:sightmachine/rocksdb (push)
+
+```
+
+## Git flow
+
+facebook:
+* work happens in multiple branches
+* branches are named `7.8.fb`
+* tags are named `v7.8.3`
+
+sightmachine:
+* work happen in only one _current_ branch, and it's NOT named `main` or `master`
+* tag `v7.8.3` will form base of branch `sm-main-7.8.3`
+* *this* will be git default branch (https://github.com/sightmachine/rocksdb/settings/branches)
+* sightmachine patches will be added on top of it.
+
+* When switching to another upstream branch ( e.g. 7.9.2)
+  * push facebook tag to sightmachine `git push sightmachine v7.9.2`
+  * branch `sm-main-7.9.2` create from upstream tag `v7.9.2`
+  * the changes from `v7.8.3...sm-main-7.8.3` cherry-picked to `sm-main-7.9.2`
```

Comment:
I think we should also add a step to review the output of
```sh
git diff v7.7.3...v7.8.3 -- CMakeLists.txt Makefile java/CMakeLists.txt java/Makefile
```
and see if any new source files need to be added to CMakeLists.txt.